### PR TITLE
Mark caveats as a production-ready feature

### PR DESCRIPTION
**Pull request by josephschorr**
Also adds the bulk lookup for caveats in the expression evaluator

### CODE REVIEW

File: internal/caveats/run.go [hunk 1]
```diff
@@ -98,13 +99,67 @@ func runExpression(
...
+	if ok {
+		return caveat, deserialized, nil
+	}
+
+	deserialized, err := caveats.DeserializeCaveat(caveat.SerializedExpression)
+	if err != nil {
+		return caveat, nil, err
+	}
+
+	lc.deserializedCaveats[caveatDefName] = deserialized
```
Comment (vroldanbet) [inline internal/caveats/run.go:149]: the assumption is that this won't get modified concurrently because the caveat tree is executed serially, right?

**Review by vroldanbet — COMMENTED** [review]
shouldn't a test have failed because of the gating being removed? 🤔

File: internal/caveats/run.go (hunk 1 above)
Comment (josephschorr) [reply internal/caveats/run.go:149]: Correct

**Comment by josephschorr** [comment]
> shouldn't a test have failed because of the gating being removed? 🤔

No, we didn't have a test for the gating

**Review by vroldanbet — APPROVED** [review]

**Pull request merged**